src/dbnode/storage/index.go

```diff
@@ -1056,12 +1056,17 @@ func (i *nsIndex) AggregateQuery(
 	defer sp.Finish()
 
 	// Get results and set the filters, namespace ID and size limit.
-	results := i.aggregateResultsPool.Get()
 	aopts := index.AggregateResultsOptions{
 		SizeLimit:   opts.Limit,
 		FieldFilter: opts.FieldFilter,
 		Type:        opts.Type,
 	}
+	var results index.AggregateResults
+	if i.aggregateResultsPool != nil {
+		results = i.aggregateResultsPool.Get()
+	} else {
+		results = index.NewAggregateResults(nil, aopts, nil)
+	}
```

Comment:
This is healthy paranoia, but I believe if we're ever not supplying the pool I'd prefer to call that a "code bug" and have it crash. Thoughts?